pyodide_build/pywasmcross.py

```diff
@@ -118,6 +118,9 @@ def replay_genargs_handle_dashl(arg: str, used_libs: set[str]) -> str | None:
     if arg == "-lgfortran":
         return None
 
+    if arg in ["-lfreetype", "-lpng"]:
+        arg += "-wasm-sjlj"
```

Comment:
@ryanking13 maybe we can gate this on abi >= 2025_0? Or what do you think is the best way of doing this only when building against Pyodide 0.28 or later?

Reply:
Or perhaps we should add a separate config variable for `WASM_EXCEPTIONS`.

Reply:
I am okay with dropping support for Pyodide <= 0.27 if introducing the ABI or Pyodide version condition makes things too complicated.

However, this code seems to be very limited, only working for freetype and png libraries. Is there any way to specify this flag in the package recipe, not in pywasmcross.py?

Reply:
Well the link args are generated by the build backend, so I don't see how. Unless they come from pkg-config or something.

Reply:
I can add a comment to look into cleaning this up.

Reply:
Yeah, we can leave it as TODO for now, but I would like to clean this up later.

Reply:
Yeah, I agree. Instead of just adding "clean this up", how about adding some context for future readers then? I think there are a bunch of hacky things in this file, which probably is now outdated. But there are not enough comments in the code about the context, so it is hard to figure out.